django_ibmi/operations.py

```diff
@@ -207,15 +178,24 @@ def date_interval_sql(self, timedelta):
             timedelta.days, timedelta.seconds, timedelta.microseconds), []
 
     # As casting is not required, so nothing is required to do in this function.
-    def datetime_cast_sql(self):
+    def datetime_cast_date_sql(self, field_name, tzname):
         return "%s"
```

Comment:
Looks like datetime_cast_date_sql differs in that it's supposed to format the string instead of returning the format string itself: https://github.com/django/django/blob/76c0b32f826469320c59709d31e2f2126dd7c505/django/db/backends/mysql/operations.py#L94-L96

We could use `CAST(%s as DATE)` to ensure the field is treated as a date. Should remove the comment above, too.